Change wording of use a generator message for sum/max/min (#6600)
PyCQA/pylint: doc/data/messages/c/consider-using-generator/details.rst

> Removing ``[]`` inside calls that can use containers or generator indistinctly should be considered for performance reasons since a generator will have an upfront cost to pay. The performance will be better if

Reviewer: ```suggestion
Removing ``[]`` inside calls that can use containers or generators should be considered for performance reasons since a generator will have an upfront cost to pay. The performance will be better if
```

> you are working with really long lists or sets and depending on the interpreter: you should consider
> your dataset and your environment.

Reviewer: ```suggestion
you are working with long lists or sets.
```

"Depending on you're interpreter" doesn't really add much imo. Documentation like this is mostly meant for beginners and without clear directions about what to then think about when they are thinking about their interpreter it doesn't add much and just confuses them. Length of the iterator is a specific tip so that works well.

> For ``max``, ``min`` and ``sum` using a generator is also best practices recommended by pep289.

Reviewer: ```suggestion
For ``max``, ``min`` and ``sum` using a generator is also recommended by pep289.
```

> Removing ``[ ]`` inside containers should be considered for performance reasons since

Author: ```suggestion
Removing ``[ ]`` inside calls that can use containers or generator indistinctly should be considered for performance reasons since
```

> Removing ``[ ]`` inside calls that can use containers or generator indistinctly should be considered for performance reasons since
> list comprehension will make unnecessary data copies. The performance will be better if
> you are working with really long lists or sets.
> 
> Regarding performance it's the same for ``max``, ``min`` and ``sum``: you should consider
> your dataset and your environment. But it's also best practice recommended by pep289.

Author: ```suggestion
Removing ``[]`` inside calls that can use containers or generator indistinctly should be considered for performance reasons since a generator will have an upfront cost to pay. The performance will be better if
you are working with really long lists or sets and depending on the interpreter: you should consider
your dataset and your environment. 

For ``max``, ``min`` and ``sum` using a generator is also best practices recommended by pep289.
```
